toqito/matrix_props/is_k_incoherent.py

```diff
@@ -0,0 +1,115 @@
+"""Checks if the matrix is $k$-incoherent."""
+
+from itertools import combinations
+
+import cvxpy as cp
+import numpy as np
+
+from toqito.matrices import comparison
+from toqito.matrix_props import is_positive_semidefinite, is_square
+
+
+def is_k_incoherent(mat: np.ndarray, k: int, tol: float = 1e-15) -> bool:
+    r"""Determine whether a quantum state is k-incoherent :cite:`Johnston_2022_Absolutely`.
+
+    For a positive integer, :math:`k`, the matrix :math:`X \in \text{Pos}(\mathbb{C}^n)` is called :math:`k`-incoherent*
+    if there exists a positive integer :math:`m`, a set  :math:`S = \{|\psi_0\rangle, |\psi_1\rangle,\ldots,
+    |\psi_{m-1}\rangle\} \subset \mathbb{C}^n` with the property that each :math:`|\psi_i\rangle` has at most :math:`k`
+    non-zero entries, and real scalars :math:`c_0, c_1, \ldots, c_{m-1} \geq 0` for which
```

Comment:
define what $n$ is here

Reply:
Done in f6636dc808310c7901354bfc3082e63794cd0567